pyminio/main.py

```diff
@@ -207,115 +262,202 @@ def rmdir(self, path: str, recursive: bool = False):
 
     def mv(self, from_path: str, to_path: str):
-        self.cp(from_path, to_path)
-        self.rm(from_path)
+        """Move files from one directory to another.
+
+        Works like linux's mv.
 
-    def get(self, path: str):
-        bucket, directory_path, filename = self._get_match(path)
-        relative_path = self._get_relative_path(directory_path, filename)
+        Args:
+            from_path: source path.
+            to_path: destination path.
+        """
+        from_match = Match(from_path)
+        to_match = Match(to_path).calculate_match(from_match)
 
         try:
+            self.cp(from_match.path, to_match.path)
 
-            data = self.minio_obj.get_object(bucket, relative_path).data
-            details = self.minio_obj.stat_object(bucket, relative_path)
-            details_metadata = self._extract_metadata(details.metadata)
+        finally:
+            if(self.exists(from_match.path) and self.exists(to_match.path)):
+                self.rm(from_match.path)
 
-            metadata = {
-                "is_dir": details.is_dir,
-                "last_modified": details.last_modified,
-                "size": details.size
-            }
-            metadata.update(details_metadata)
+    def get(self, path: str) -> ObjectData:
+        """Get file or directory from minio.
 
-            return File(name=filename, full_path=path, data=data,
-                        metadata=AttrDict(metadata))
+        Args:
+            path: path of a directory or a file.
+        """
+        match = Match(path)
+        kwargs = AttrDict()
 
-        except NoSuchKey:
+        if match.is_bucket():
+            raise ValueError('Minio bucket has no representable object.')
+        try:
+            if match.is_file():
+                kwargs.data = self.minio_obj.get_object(
+                    match.bucket, match.relative_path).data
+
+                details = self.minio_obj.stat_object(
+                    match.bucket, match.relative_path)
+                name = match.filename
+                return_obj = File
+
+            else:
+                parent_directory = \
+                    join(dirname(normpath(match.prefix)), '')
+                objects = self.minio_obj.list_objects(
+                    bucket_name=match.bucket,
+                    prefix=parent_directory
+                )
+
+                details = next(filter(
+                    lambda obj: obj.object_name == match.relative_path,
+                    objects))
+                name = join(basename(normpath(details.object_name)), '')
+                return_obj = Folder
+
+        except (NoSuchKey, StopIteration):
             raise RuntimeError(f"cannot access {path}: "
                                "No such file or directory")
 
-    def put_data(self, path: str, data: Union[str, File],
+        details_metadata = \
+            self._extract_metadata(details.metadata)
+
+        metadata = {
+            "is_dir": details.is_dir,
```

Comment:
Where `is_dir` was added to the details? I can't find it

Reply:
minio's stat_object return data like is_dir and created_at modified_at and more.
list_objects add this data to the object aswell.